src/filters/filter_shift.hpp

```diff
@@ -127,7 +143,8 @@ namespace dmxfish::filters {
                     values.at(i) = values.at(i-1);
                 }
                 values.at(0) = *input;
-                last_update = last_update + *switch_time;
+                //last_update = last_update + *switch_time;
+		last_update = *time;
```

Comment:
the old version was for a predictable long duration. Now with every update it waits a bit longer than  switch time.